From fdcd2a57f4dba25d1a0bb5c8b4f36c1f7f949137 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Date: Thu, 8 Aug 2019 15:04:08 +0200
Subject: [PATCH] irqchip/gic-v3-its: fix build warnings
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Patch-mainline: Never, Build warnings due to missing patch
References: bsc#1144880

Following warnings are fixed:
drivers/irqchip/irq-gic-v3-its.c: In function ‘its_irq_domain_activate’:
drivers/irqchip/irq-gic-v3-its.c:2323:11: warning: ‘return’ with a value, in function returning void
    return -EINVAL;
           ^
drivers/irqchip/irq-gic-v3-its.c:2307:13: note: declared here
 static void its_irq_domain_activate(struct irq_domain *domain,
             ^~~~~~~~~~~~~~~~~~~~~~~
  LD      drivers/net/ethernet/allwinner/built-in.o
drivers/irqchip/irq-gic-v3-its.c: In function ‘its_vpe_irq_domain_activate’:
drivers/irqchip/irq-gic-v3-its.c:2838:10: warning: ‘return’ with a value, in function returning void
   return 0;
          ^
drivers/irqchip/irq-gic-v3-its.c:2830:13: note: declared here
 static void its_vpe_irq_domain_activate(struct irq_domain *domain,

Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/irqchip/irq-gic-v3-its.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/irqchip/irq-gic-v3-its.c b/drivers/irqchip/irq-gic-v3-its.c
index fbda05bbd9ee..03ace8f3ac65 100644
--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -2319,8 +2319,10 @@ static void its_irq_domain_activate(struct irq_domain *domain,
 	/* Bind the LPI to the first possible CPU */
 	cpu = cpumask_first_and(cpu_mask, cpu_online_mask);
 	if (cpu >= nr_cpu_ids) {
-		if (its_dev->its->flags & ITS_FLAGS_WORKAROUND_CAVIUM_23144)
-			return -EINVAL;
+		if (its_dev->its->flags & ITS_FLAGS_WORKAROUND_CAVIUM_23144) {
+			pr_err("ITS: Can't bind LPI to non-local node CPU due to Cavium erratum 23144\n");
+			return;
+		}
 
 		cpu = cpumask_first(cpu_online_mask);
 	}
@@ -2835,7 +2837,7 @@ static void its_vpe_irq_domain_activate(struct irq_domain *domain,
 
 	/* If we use the list map, we issue VMAPP on demand... */
 	if (its_list_map)
-		return 0;
+		return;
 
 	/* Map the VPE to the first possible CPU */
 	vpe->col_idx = cpumask_first(cpu_online_mask);
-- 
2.11.0

